Moved session into PostgreSQL

asbjornu opened the pull request:
This PR is not necessarily created for you to merge, but more to let you know of the work I've been doing to get Drywall up and running. What I ended up doing, which I think is a good idea anyway, was to have the session stored in PostgreSQL, which I achieved through [connect-session-sequelize](/connect-session-sequelize).

I've also done a couple of other things (like renaming the cookie) you don't want to merge, but if any of this looks interesting, please let me know so I can bisect the interesting (to you) commits into a branch that might be easier to merge.

So please **do not merge** this, but have a look around and let me know if there's anything you like.

jedireza committed:
updated less variables file for bootstrap v3.3.5

jedireza committed:
Merge pull request #295 from jedireza/bootstrap-build

updated less variables file for bootstrap v3.3.5

jedireza committed:
version bumped 0.16.2

benmarten committed:
Fixed HTML5 Validator Errors

jedireza committed:
Merge pull request #301 from benmarten/validHTML5

Fixed HTML5 Validator Errors

jedireza committed:
version bumped 0.16.3

jedireza committed:
update deps

jedireza committed:
Merge pull request #306 from jedireza/update-deps

update deps

jedireza committed:
version bumped 0.16.4

jedireza committed:
added link to email trouble wiki page

jedireza committed:
Merge pull request #310 from jedireza/readme-email-trouble

added link to email trouble wiki page

asbjornu committed:
Ignore config/config.json

asbjornu committed:
Added missing cp statement to readme

asbjornu committed:
Removed unused mongodb property from config.js.tmpl

asbjornu committed:
Added PostgreSQL configuration and updated a few things here and there

asbjornu committed:
Translated a bunch of stuff to Norwegian. Nowhere near complete.

asbjornu committed:
Replaced logos with Pirate flags

asbjornu committed:
More translation from English to Norwegian. This time with URLs in routes.js, so stuff actually works. Not at all done yet.

asbjornu committed:
Replaced favicon with pirate flag

tobixen committed:
s|/registrering|/signup| - effectively reverting some of ef9047093e77f5ecb9b69e40f14c55de53d5d633 - for one thing, this URL translation idea break things, for the other thing, s|/registrering|/signup| seems to involve less changes than s|/signup|/registrering| and for the third thing, IMHO, /signup doesn't look that bad anyway

tobixen committed:
Misc search&replace, effectively reverting the rest of the URL-translations in ef9047093e77f5ecb9b69e40f14c55de53d5d633

asbjornu committed:
More translation

asbjornu committed:
Signup should now be 'registrering' everywhere

asbjornu committed:
Login is now 'logg-inn'

asbjornu committed:
Account is now 'konto'

asbjornu committed:
`req.session` might be null.

For some reason I haven’t figured out yet, `req.session` is null, which
causes the exception “can’t set returnUrl of undefined”. Checking if
`req.session` is defined therefore feels like a good idea.

asbjornu committed:
Don't drop the database on every restart

asbjornu committed:
Ignore log files

asbjornu committed:
Add connect-pg-simple and use it for session management instead of Redis

asbjornu committed:
Console logging all the things to figure out why login doesn't work.

asbjornu committed:
Uninstall connect-redis

asbjornu committed:
Added missing browser-logger dev dependency

asbjornu committed:
Moved the "force" Sequelize option to config.json

asbjornu committed:
Replaced connect-pg-simple with connect-session-sequelize.

asbjornu committed:
Change the cookie name from the default 'connect.sid' to 'piratpartiet.sid'

asbjornu committed:
Upgrade all the things

asbjornu committed:
Improved logging

asbjornu committed:
Updated documentation

trystant commented:
Hi, first of all, thanks for this work. I'm sorry for the slow response, but I've just started a new job and that's affected things for the time being. I'll take a look and let you know over the next few days.

asbjornu committed:
More uniform and extensive logging in the signup and login workflow

asbjornu committed:
More duplicate username logging

asbjornu committed:
Since we can't set 'console.log' as a value in config.json (since it's JSON and not full JavaScript), we need to set 'console.log' if 'logging' evaluates to true in models/index.js.

asbjornu committed:
Seems like after upgrading Sequelize, the `find` methods need a `where` property.

asbjornu committed:
Fixed queries using a `conditions` variable.

asbjornu committed:
More logging in passport.js

asbjornu committed:
Set res.locals to itself or an object.

asbjornu committed:
Disable browser logger since it apparently tries to set headers after they are sent.

asbjornu committed:
Improved logging in passport.js

asbjornu committed:
Downgrade Passport to v0.2.2 due to jaredhanson/passport#400

asbjornu committed:
I quite like literal object syntactic sugar.

asbjornu committed:
Rewrote the `generateToken` and `patchUser` workflows to use Sequelize instead of Mongoose. Seems like there's a lot of traces of Mongoose left in the codebase. Sigh. :(

asbjornu committed:
Remove /config/config.json, move its config into /config.js and replace the usage around the block.

asbjornu committed:
The `sendMail` workflow was b0rked. End with `response` instead of `validate` to avoid "Error: Can't set headers after they are sent".

asbjornu committed:
Moved log statement to the correct place

asbjornu committed:
Replace Mongoose with Sequelize in the reset password view

asbjornu committed:
Go to /account/ after successfully logging in

asbjornu committed:
Simplify the reset logic. As we already have the `user`, we can just save() it in place.

asbjornu committed:
Switch the `if (!user)` statement around and improve logging.

asbjornu committed:
Improve reset logging and added catch promise

asbjornu committed:
Updated logging config

asbjornu committed:
Since we're not forced to JSON syntax anymore, remove quotes for readability. Also updated readme.

tobixen committed:
lower case table names.  but still attribute names are in camelCase

asbjornu committed:
Updated mocha from v2.3.0 to v2.3.4

asbjornu committed:
Most of the codebase is indented with 2 spaces, so let's standardize on that.

asbjornu committed:
Move app setup into an exported function that can be invoked in tests

asbjornu committed:
Rewrite the index spec so it actually works

tobixen committed:
s/user/login

tobixen committed:
account table should contain most of the data we need about our members

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

asbjornu committed:
Remove unused logging config

tobixen committed:
account table should contain most of the data we need about our members

asbjornu committed:
Run tests on port 3001 and configure the test environment database differently from development.

Commit:
bugfixing

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

asbjornu committed:
Add more node js versions to Travis

tobixen committed:
Revert "s/user/login"

This reverts commit a66be22d5b28cf555d9f9b522d1479c981b6b7d5.

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

asbjornu committed:
Removing old versions of node js from travis

asbjornu committed:
Remove grunt from dependencies, since it's a devDependency

tobixen committed:
revert db.user back to db.User

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

asbjornu committed:
Updated chai from 3.2 to 3.4.1

asbjornu committed:
Updated grunt

asbjornu committed:
Updated index spec metadata and documentation

asbjornu committed:
Added (currently failing) signup spec

asbjornu committed:
Handle failing db models

asbjornu committed:
Pass the csrf token for signup

asbjornu committed:
Use send() to send POST data

asbjornu committed:
Increase the signup timeout to 5 seconds

asbjornu committed:
Added stuff from https://github.com/redis/hiredis-node/blob/master/.travis.yml to Travis to make it work, hopefully.

asbjornu committed:
Added before_script to Travis to copy config.js.tmpl to config.js

asbjornu committed:
Added database config and setup to Travis

asbjornu committed:
Add Travis db commands to drop role and database if 'drywall' already exist

asbjornu committed:
Updated readme with correct psql commands

asbjornu committed:
Updated Travis with correct psql commands

asbjornu committed:
Added build status badge and updated dependency and devDependency status badges to point to piratpartiet's fork.

asbjornu committed:
Added IRC build notification to Travis

asbjornu committed:
Implemented app.utility.debug and .error functions and introduced config.logging object so logging can be turned on or off for debug and error messages globally.

tobixen committed:
revert lower case table names - with conflict handling

tobixen committed:
let's try to use those cool sequalize table configuration parameters - timestamp, underscored, unfrozen tablename - plus, the user table is renamed to login_user

tobixen committed:
oups, silly-stupid bugfix

tobixen committed:
Minor bugfix - s/table_name/tableName/

tobixen committed:
s/Account/Member/ - this table contains data we need or may want from our members

tobixen committed:
s/Account/Member/ - this table contains data we need or may want from our members

tobixen committed:
s/Account/Member/ - this table contains data we need or may want from our members

tobixen committed:
tableName all over the place to ensure underscore tablenames

tobixen committed:
lower_case attribute names in the db layer

tobixen committed:
stupid silly comment

tobixen committed:
s/\t/        /

tobixen committed:
how to make comments in .js?

tobixen committed:
Payment table - for membership fees paid, donations and other purposes.  Designed to be ready for Crypto-currencies.

asbjornu committed:
Removed the "-spec" name from specs

tobixen committed:
silly stupid syntax error fix

tobixen committed:
Minor bugfix - s/table_name/tableName/

asbjornu committed:
Fixed bugs from 40597153d8dbb1afb652564126d3dfea0b98377f in the config template

tobixen committed:
minor consistency-fix - s/receipient_address/receipientAddress/

tobixen committed:
it didn't roll out the payment table

tobixen committed:
silly stupid syntax error fix

tobixen committed:
forgot tableName on the new payments table

asbjornu committed:
Added config specs to verify that config.js is in sync with config.js.tmpl

asbjornu committed:
Changed the IRC template for Travis to be a bit more informative. Also try to "skip_join" of the bot.

asbjornu committed:
Seems like "skip_join" didn't work

asbjornu committed:
Condense the IRC message somewhat

asbjornu committed:
Condense the IRC notification even more

asbjornu committed:
Change the order of the IRC notification variables a bit to improve readability.

Commit:
Update irc output.

tobixen committed:
RoleAssignment: to give people positions, privileges and admin rights

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

tobixen committed:
forgotten file in prev commit

tobixen committed:
bugfix - stupid silly cut'n'paste-error

asbjornu committed:
Ignore *config.js and .json so it's possible to have copies of the config file without checking it in

asbjornu committed:
Increase timeout

asbjornu committed:
Added logging module so the global logging can be configured properly and independently for both database and regular application-level logging.

asbjornu committed:
Beautified and strict

asbjornu committed:
Removed duplicate name property and its corresponding comment

asbjornu committed:
The log method is called log(), not debug().

asbjornu committed:
Implement User.canPlayRoleOf() through lookup in RoleAssignment

tobixen committed:
syncing script between old db and new db

tobixen committed:
Merge branch 'master' into sync-script

tobixen committed:
isVerified -> is_verified

tobixen committed:
Merge branch 'master' into sync-script

asbjornu committed:
Don't crash if config is undefined; just treat it as if logging is enabled and log an error about it.

asbjornu committed:
Fix logout so it actually works

asbjornu committed:
Remove the requirement that a user needs to have the "account" role to view his or her account.

Commit:
Merge https://github.com/piratpartiet/drywall

tobixen committed:
we need municipality and county

tobixen committed:
attempt on importing historic membership fees in to the drywall db

tobixen committed:
Merge branch 'master' into sync-script

asbjornu committed:
Added login spec and attempt to use the signup spec within the login spec. Doesn't work, so I'll revert.

asbjornu committed:
Revert the module-ification of the signup spec from commit 6b86b4923a8feaf1407797f36b7dc91bd934feed

tobixen committed:
NOT TESTED and hopefully we won't need it much - but, functionality to sync new members coming into the new system into the old db

asbjornu committed:
Added uuid and printf packages

asbjornu committed:
Change signup spec to use random username and e-mail address

asbjornu committed:
Fix login test by creating a randomly named user to login with

asbjornu committed:
Formatting

asbjornu committed:
Expect 'set-cookie' to exist in the headers collection.

asbjornu committed:
Removed unused require statements

asbjornu committed:
Throw all errors occurring in specs

asbjornu committed:
Added account tests by making login into a module that exports a login() function which through superagent sticks cookies to the requests and makes everything work like magic.

asbjornu committed:
Expect the Location header to equal /login/

asbjornu committed:
The foreign key for User in the Member table is called `user_id`.

asbjornu committed:
Better handling of null account.name

tobixen committed:
Merge branch 'master' into pirate-party-customizations

Conflicts:
	models/User.js
	public/views/login/index.js
	routes.js
	views/login/index.js
	views/signup/index.js

asbjornu committed:
Beautify

asbjornu committed:
Account is renamed to Member, so we need to propagate that all the way up to Backbone.js. Adds a test that verifies this and fixes #2 along with 5cc7d3b3160ffd0fe0ddcae30eef941e1195090f.

tobixen committed:
undoing some more url-translations

tobixen committed:
undoing some more url-translations

tobixen committed:
l10n av tester

tobixen committed:
mer l10n av tester

tobixen committed:
Merge branch 'sync-script' into pirate-party-customizations

Commit:
Merge https://github.com/piratpartiet/drywall

asbjornu committed:
Add sequelizeStore to app.db and set max listeners to 0 in the tests to avoid:

warning: possible EventEmitter memory leak detected. 11 disconnect listeners added. Use emitter.setMaxListeners() to increase limit.
    at SequelizeStore.addListener (events.js:239:17)
    at session (./node_modules/express-session/index.js:150:9)
    at Object.exports.setup (./app.js:46:11)
    at Context.<anonymous> (./specs/signup.js:17:22)
    at callFn (Drywall/node_modules/mocha/lib/runnable.js:286:21)
    at Hook.Runnable.run (./node_modules/mocha/lib/runnable.js:279:7)
    at next (./node_modules/mocha/lib/runner.js:297:10)
    at Immediate._onImmediate (./node_modules/mocha/lib/runner.js:319:5)
    at processImmediate [as _immediateCallback] (timers.js:383:17)

asbjornu committed:
Extract user creation to a reusable createUser() function

asbjornu committed:
Add forgot password specs

asbjornu committed:
Removed console.log

asbjornu committed:
Extensive logging in sendmail and forgot password to figure out why it doesn't work as per #3.

asbjornu committed:
Aha. We need to invoke the correct workflow. Fixes #3.

tobixen committed:
Merge branch 'master' into pirate-party-customizations

 Conflicts:
	specs/login.js
	specs/signup.js

tobixen committed:
yet another L10n problem with test code

tobixen committed:
bugfixing

tobixen committed:
Merge branch 'sync-script-tmp-test' into pirate-party-customizations-gh

tobixen committed:
Merge pull request #4 from piratpartiet/pirate-party-customizations-gh

Pirate party customizations gh

tobixen committed:
fix for silly stupid search'n'replace mistakes

asbjornu committed:
Start PostgreSQL with `npm start`

asbjornu committed:
Since contact details might be missing, Backbone thinks it's a new resource and wants to do POST, but we don't have a route matching POST, so we have to force HTTP PUT when saving Details.

asbjornu committed:
Add a failing test to update contact details and adjust stuff in the client and server views to reflect reality a bit better

asbjornu committed:
Output the response text if the parsed JSON object's success property isn't true.

asbjornu committed:
Fix the patchAccount workflow so it can work. Still not there, but closer.

asbjornu committed:
Removed member.name.full assignment.

tobixen committed:
upstart script for our ubuntu server

tobixen committed:
README og litt tabellforandringer

tobixen committed:
bugfix - missing comma

tobixen committed:
Bugfix: missing comma

tobixen committed:
oups, forgot to add the README, plus tweaking the member table

tobixen committed:
Merge branch 'master' of github.com:piratpartiet/drywall

tobixen committed:
drywall does not accept the README.md here.  will the github markdown parser break down and cry due to the comment-out trick?

tobixen committed:
ignore the README.md file when reading directory

tobixen committed:
now we don't need those /* comment signs */ in the README anymore

garfbargle committed:
Using bootstraps' variables file at public/vendor/bootstrap/less/variables.less, instead of our static copy at public/less/bootstrap-vars.less

jedireza committed:
Merge pull request #336 from garfbargle/master

Import bootstrap's variables file into boostrap-build.less

jedireza committed:
0.16.5

garfbargle committed:
Replaced all imports of 'public/less/bootstrap-vars.less' with import of '/public/vendor/bootstrap/less/variables.less'

jedireza committed:
Merge pull request #338 from garfbargle/master

Using vendor/boostrap/variables.less throughout application.

jedireza committed:
0.16.6

asbjornu committed:
Update .travis.yml

Added [Codecov][codecov] to Travis.

[codecov]: https://codecov.io/github/piratpartiet/drywall

asbjornu committed:
Require sudo

Codecov uses `pip` which seems to require sudo. Let's try to enable it.

asbjornu committed:
Force Bootstrap to version 3.3.5

Force Bootstrap to version 3.3.5 instead of 3.x.x. Should fix #1.

asbjornu committed:
Ignore code coverage reports

asbjornu committed:
Let's try code coverage without sudo

asbjornu committed:
Remove iojs and add Node.js 4.2 from Travis

asbjornu committed:
Added istanbul 0.4.2 as a dev dependency

asbjornu committed:
Run istanbul cover on test

On `npm test`  run `istanbul cover` with `mocha` to both execute the tests and get code coverage at the same time

asbjornu committed:
Only run code coverage on Travis

asbjornu committed:
Move NODE_ENV to the env section

asbjornu committed:
Run istanbul as the build script instead of after success

asbjornu committed:
Combine environment variables

gitter-badger committed:
Add Gitter badge

asbjornu committed:
Added codecov badge

asbjornu committed:
The branch needs to be `codecov` for now

asbjornu committed:
Add a few sections to the readme

asbjornu committed:
Merge pull request #24 from piratpartiet/codecov

Implement code coverage status with Codecov

asbjornu committed:
Set the codecov branch to `master`

asbjornu committed:
Merge remote-tracking branch 'gitter-badger/gitter-badge-1'

* gitter-badger/gitter-badge-1:
  Add Gitter badge
  0.16.6
  Replaced all imports of 'public/less/bootstrap-vars.less' with import of '/public/vendor/bootstrap/less/variables.less'
  0.16.5
  Using bootstraps' variables file at public/vendor/bootstrap/less/variables.less, instead of our static copy at public/less/bootstrap-vars.less
  added link to email trouble wiki page
  version bumped 0.16.4
  update deps
  version bumped 0.16.3
  Fixed HTML5 Validator Errors
  version bumped 0.16.2
  updated less variables file for bootstrap v3.3.5

# Conflicts:
#	README.md
#	package.json
#	public/less/bootstrap-vars.less
#	views/index.jade

asbjornu committed:
Merge remote-tracking branch 'codersArmy/master'

* codersArmy/master:
  0.16.6
  Replaced all imports of 'public/less/bootstrap-vars.less' with import of '/public/vendor/bootstrap/less/variables.less'
  0.16.5
  Using bootstraps' variables file at public/vendor/bootstrap/less/variables.less, instead of our static copy at public/less/bootstrap-vars.less
  added link to email trouble wiki page
  version bumped 0.16.4
  update deps
  version bumped 0.16.3
  Fixed HTML5 Validator Errors
  version bumped 0.16.2
  updated less variables file for bootstrap v3.3.5

# Conflicts:
#	package.json
#	public/less/bootstrap-vars.less
#	views/index.jade

asbjornu committed:
Merge branch 'gitter' into upstream-merge

* gitter:
  Add Gitter badge

# Conflicts:
#	package.json

asbjornu committed:
Zip codes aren't integers

asbjornu committed:
getMember() returns a Promise, so treat it like a Promise. Also added more logging and error handling.

asbjornu committed:
Less indentation by one-lining findById().then()

asbjornu committed:
More logging

asbjornu committed:
Consistently name all name fields with a `Name` suffix.

Rename `first`, `middle` and `last` to `firstName`, `middleName` and `lastName` and ensure that they are synced from the right place in the client JavaScript.

tobixen committed:
untested stuff

asbjornu committed:
No need to re-fetch the user from the database as we already have a user on the request.

asbjornu committed:
If the user has no member, create it

asbjornu committed:
Remove console.log

asbjornu committed:
Merge pull request #28 from piratpartiet/account-settings

Fix saving of account info in settings

asbjornu committed:
Formatting

asbjornu committed:
Added failing change password spec

Added failing change password spec and changed the title of the update contact info spec.

asbjornu committed:
Remove `models`

Remove `models` from the object path

asbjornu committed:
Fix password update

Fix the `patchUser` workflow in account settings password so the password can be updated. Fixes #8.

asbjornu committed:
Merge pull request #29 from piratpartiet/feature/change-password

Fix change password

asbjornu committed:
Merge remote-tracking branch 'trystant/master' into feature/merge-trystant-master

* trystant/master: (28 commits)
  Updated Postgres system requirement for #22
  Updated README.md installation process
  Updated README.md with testing info
  Updated README to remove supertest
  Added codeclimate test coverage badge; #20
  Changed Markdown version to HTML for README.md #20
  Updated README.md and removed unused code; #20
  Changed lcov.info location for code climate;  #20
  Added commands to generate lcov data; #20
  Added config variables to help test coverage; #20
  Added istanbul & configured for test coverage; #20
  Added Code Climate badge; #20
  Added CodeClimate badge to the README.md; #20
  Remove console.log from AccountSpec; #10
  Added diagnostics to account spec #10
  Changed "127.0.0.1" to "localhost"; #10
  Manually create database; #10
  Quoted replacement values
  Changed echo to cat #10
  Updated script to show template output; #10
  ...

# Conflicts:
#	.travis.yml
#	README.md
#	app.js
#	config/config.json.tmpl
#	models/Account.js
#	package.json

asbjornu committed:
Added a couple of identity specs

asbjornu committed:
Use member instead of account

Rename `patchAccount` to `patchMember` for clarity

asbjornu committed:
Fix identity update

Fix updating the identity (username and/or e-mail address). Fixes #7.

asbjornu committed:
Merge pull request #30 from piratpartiet/feature/update-identity

Fix update identity

asbjornu committed:
Merge branch 'master' into feature/merge-trystant-master

* master:
  Fix identity update
  Use member instead of account
  Added a couple of identity specs

asbjornu committed:
Merge pull request #31 from piratpartiet/feature/merge-trystant-master

Merge trystant master

trystant commented:
Sorry for not having followed up on this before. I'll be reviewing this [PR 15](https://github.com/trystant/drywall/pull/18) and  over the next week & incorporating changes. Glad to see your work on this @asbjornu and everyone.

asbjornu commented:
@trystant Take your time. :smiley: :+1:
